tests/splines/batched_2d_spline_builder.cpp

```diff
@@ -215,14 +239,14 @@ static void Batched2dSplineTest()
 
     // Create the values domain (mesh)
 #if defined(BC_HERMITE)
-    auto interpolation_domain1
-            = ddc::DiscreteDomain<IDim<I1, I1, I2>>(GrevillePoints<BSplines<I1>>::get_domain());
-    auto interpolation_domain2
-            = ddc::DiscreteDomain<IDim<I2, I1, I2>>(GrevillePoints<BSplines<I2>>::get_domain());
+    auto interpolation_domain1 = ddc::DiscreteDomain<IDim<I1, I1, I2>>(
+            GrevillePoints<IDim<I1, I1, I2>, BSplines<I1>>::get_domain());
+    auto interpolation_domain2 = ddc::DiscreteDomain<IDim<I2, I1, I2>>(
+            GrevillePoints<IDim<I2, I1, I2>, BSplines<I2>>::get_domain());
 #endif
     auto interpolation_domain = ddc::DiscreteDomain<IDim<I1, I1, I2>, IDim<I2, I1, I2>>(
-            GrevillePoints<BSplines<I1>>::get_domain(),
-            GrevillePoints<BSplines<I2>>::get_domain());
+            GrevillePoints<IDim<I1, I1, I2>, BSplines<I1>>::get_domain(),
+            GrevillePoints<IDim<I2, I1, I2>, BSplines<I2>>::get_domain());
```

Comment:
@blegouix line 255 of this file does not compile after the change and I don't understand the logic with all the `IDim<I1, I1, I2>`. Do you know what is happening ? I think we will need to simplify this part, it is way too difficult to read for a unit test.